packages/grafana-ui/src/components/Table/types.ts

```diff
@@ -84,9 +80,31 @@ export interface Props {
   footerOptions?: TableFooterCalc;
   footerValues?: FooterItem[];
   enablePagination?: boolean;
-  cellHeight?: TableCellHeight;
+  cellHeight?: schema.TableCellHeight;
   /** @alpha */
   subData?: DataFrame[];
   /** @alpha Used by SparklineCell when provided */
   timeRange?: TimeRange;
 }
+
+export interface CustomCellRendererProps {
+  field: Field;
+  index: number;
+  frame: DataFrame;
+  // Would be great to have generic type for this but that would need having a generic DataFrame type where the field
+  // types could be propagated here.
+  value: unknown;
+}
+
+// Can be used to define completely custom cell contents by providing a custom cellComponent.
+export interface TableCustomCellOptions {
+  cellComponent: FC<CustomCellRendererProps>;
```

Comment:
Given my comment I think we should rename this property to something that can express both component and an ID. I would suggest maybe just `cell`?


Or alternatively - following up on my comment mentioned above - the grafana/schema type could be:

```
interface TableCustomCellOptions {
  type: TableCellDisplayMode.Custom;
  cellId: string; // From schema POV this is an ID of a custom cell
}
```

and the grafana/ui Table's component type:

```ts
import { TableCustomCellOptions as RawTableCustomCellOptions } from '@grafana/schema';

interface TableCustomCellOptions extends Omit<RawTableCustomCellOptions, 'cellId'> {
  type: schema.TableCellDisplayMode.Custom;
  cellComponent?:  FC<CustomCellRendererProps>;
  cellId?: RawTableCustomCellOptions['cellId'] 
}
```

I'm on a fence which one is better. Maybe having a union type is better than extending interface with additional properties `cellComponent`.

Reply:
So I wonder if this is needed. Right now I extend the `TableCellOptions` with `TableCustomCellOptions` inside the `grafana/ui` and don't touch the `grafana/schema`. Once this is needed for panel I think `TableCustomCellOptions` can be created in `grafana/schema` (or table panel) and those two should be separate exclusive types imho. Basically, there is some base in `grafana/schema` that is common but then each extends it in its own way to do custom rendering. Otherwise, I feel we would leak the table panel abstractions into table component.

Reply:
Honestly, I feel like we shouldn't even be importing `grafana/schema` into `grafana/ui` and should just live with some duplication as even though the types may be same textually they represent slightly different concepts and have different constraints as seen here.

Reply:
@aocenas - I am not suggesting changing the type in grafana/schema now - I think we will do this when we decide to provide a custom cell rendering possibility via field configuration. So for now, this is all good.